mmv1/third_party/terraform/data_sources/data_source_alloydb_supported_database_flags.go.erb

```diff
@@ -0,0 +1,242 @@
+		supportedDatabaseFlagIterator = locationsClient.ListSupportedDatabaseFlags(config.context, locsReq, gax.WithPath(url))
+		return nil
+	}, 5)
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Supported Database flags %q", d.Id()))
+	}
+
+	var supportedDatabaseFlags []map[string]interface{}
+	for {
+		supportedDatabaseFlag := make(map[string]interface{})
+		flag, err := supportedDatabaseFlagIterator.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf(fmt.Sprintf("Failed to fetch the supported database flags for the provided location: %s", location))
+		}
+		if flag.Name != "" {
+			supportedDatabaseFlag["name"] = flag.Name
+		}
+		if flag.FlagName != "" {
+			supportedDatabaseFlag["flag_name"] = flag.FlagName
+		}
+		supportedDatabaseFlag["value_type"] = flattenValueType(flag.ValueType)
+		supportedDatabaseFlag["accepts_multiple_values"] = flag.AcceptsMultipleValues
+		supportedDatabaseFlag["requires_db_restart"] = flag.RequiresDbRestart
+		if flag.SupportedDbVersions != nil {
+			dbVersions := make([]string, 0, len(flag.SupportedDbVersions))
+			for _, supDbVer := range flag.SupportedDbVersions {
+				dbVersions = append(dbVersions, supDbVer.String())
+			}
+			supportedDatabaseFlag["supported_db_versions"] = dbVersions
+		}
+
+		if flag.Restrictions != nil {
+			if stringRes, ok := flag.Restrictions.(*alloydbpb.SupportedDatabaseFlag_StringRestrictions_); ok {
+				restrictions := make([]map[string][]string, 0, 1)
+				fetchedAllowedValues := stringRes.StringRestrictions.AllowedValues
+				if fetchedAllowedValues != nil {
+					allowedValues := make([]string, 0, len(fetchedAllowedValues))
+					for _, val := range fetchedAllowedValues {
+						allowedValues = append(allowedValues, val)
+					}
+					stringRestrictions := map[string][]string{
+						"allowed_values": allowedValues,
+					}
+					restrictions = append(restrictions, stringRestrictions)
+					supportedDatabaseFlag["string_restrictions"] = restrictions
+				}
+			}
+			if integerRes, ok := flag.Restrictions.(*alloydbpb.SupportedDatabaseFlag_IntegerRestrictions_); ok {
+				restrictions := make([]map[string]int64, 0, 1)
+				minValue := integerRes.IntegerRestrictions.MinValue
+				maxValue := integerRes.IntegerRestrictions.MaxValue
+				integerRestrictions := map[string]int64{
+					"min_value": minValue.GetValue(),
+					"max_value": maxValue.GetValue(),
+				}
+				restrictions = append(restrictions, integerRestrictions)
+				supportedDatabaseFlag["integer_restrictions"] = restrictions
+			}
+		}
+		supportedDatabaseFlags = append(supportedDatabaseFlags, supportedDatabaseFlag)
+	}
+	if err := d.Set("supported_database_flags", supportedDatabaseFlags); err != nil {
+		return fmt.Errorf("Error setting supported_database_flags: %s", err)
+	}
+	d.SetId(fmt.Sprintf("projects/%s/locations/%s/supportedDbFlags", project, location))
+	return nil
+}
+
+func flattenValueType(value alloydbpb.SupportedDatabaseFlag_ValueType) string {
```

Comment:
Whats going on here? Is this not a standard enum in the service itself? Why are we using an int switch statement here?